source/crud/tests/v2/default-write-concern.yml

```diff
@@ -0,0 +1,413 @@
+                  _id: 1
+                u:
+                  x: 2
+            writeConcern: null
+      - command_started_event:
+          command:
+            insert: default_write_concern_coll
+            documents:
+              - _id: 3
+            writeConcern: null
+      - command_started_event:
+          command:
+            update: default_write_concern_coll
+            updates:
+              - q:
+                  _id: 1
+                u:
+                  $set:
+                    x: 3
+                multi: true
+            writeConcern: null
+      - command_started_event:
+          command:
+            delete: default_write_concern_coll
+            deletes:
+              - q:
+                  _id: 3
+                limit: 1
+            writeConcern: null
+  - description: 'findOneAnd{update,delete,replace}'
+    operations:
+      - name: findOneAndUpdate
+        arguments:
+          filter:
+            _id: 1
+          update:
+            $set:
+              x: 1
+      - name: findOneAndReplace
+        arguments:
+          filter:
+            _id: 2
+          replacement:
+            x: 2
+      - name: findOneAndDelete
+        arguments:
+          filter:
+            _id: 2
+    outcome:
+      collection:
+        name: default_write_concern_coll
+        data:
+          - _id: 1
+            x: 1
+    expectations:
+      - command_started_event:
+          command:
+            findAndModify: default_write_concern_coll
+            query:
+              _id: 1
+            update:
+              $set:
+                x: 1
+            writeConcern: null
+      - command_started_event:
+          command:
+            findAndModify: default_write_concern_coll
+            query:
+              _id: 2
+            update:
+              x: 2
+            writeConcern: null
+      - command_started_event:
+          command:
+            findAndModify: default_write_concern_coll
+            query:
+              _id: 2
+            remove: true
+            writeConcern: null
+  - description: 'insert{one,many}'
+    operations:
+      - name: insertOne
+        arguments:
+          document:
+            _id: 3
+      - name: insertMany
+        arguments:
+          documents:
+            - _id: 4
+            - _id: 5
+    outcome:
+      collection:
+        name: default_write_concern_coll
+        data:
+          - _id: 1
+            x: 11
+          - _id: 2
+            x: 22
+          - _id: 3
+          - _id: 4
+          - _id: 5
+    expectations:
+      - command_started_event:
+          command:
+            insert: default_write_concern_coll
+            documents:
+              - _id: 3
+            writeConcern: null
+      - command_started_event:
+          command:
+            insert: default_write_concern_coll
+            documents:
+              - _id: 4
+              - _id: 5
+            writeConcern: null
+  - description: mapReduce
```

Comment:
Given that mapReduce, runCommand, createIndex, and dropIndex are not actually defined in the CRUD spec, I wonder if we should relocate the tests in this PR to a new sub-directory in https://github.com/mongodb/specifications/tree/master/source/read-write-concern/tests.

I think the intention of [SPEC-1352](https://jira.mongodb.org/browse/SPEC-1352) was not to test that drivers omit a `writeConcern` option by default, but that they specifically detect an empty write concern provided at the client/database/collection level and then omit it in outgoing commands. To actually test that, I think you'd want to make use of the `clientOptions` or `collectionOptions` keys defined in the [transaction spec test format](https://github.com/mongodb/specifications/blob/master/source/transactions/tests/README.rst). Note that CRUD doesn't define those keys at all.

If you were to move these tests over to the write concern spec, then you could also address this by some small copypasta referencing the transaction spec test format, as is done in https://github.com/mongodb/specifications/blob/master/source/sessions/tests/README.rst#test-format. That would keep your new test README quite small.

Reply:
I like that idea. I added a section to read-write-concern/tests/README.rst describing the spec tests.